test/unit/files/test_posix.py

```diff
@@ -357,29 +459,57 @@ def _assert_user_access_granted(file_sources, user_context):
     assert_realizes_as(file_sources, "gxfiles://test1/a", "a\n", user_context=user_context)
 
 
-def _configured_file_sources(
-    include_allowlist=False, plugin_extra_config=None, per_user=False, writable=None, allow_subdir_creation=True
-) -> TestConfiguredFileSources:
-    tmp, root = setup_root()
+def _configured_file_sources_with_root(
+    include_allowlist=False,
+    plugin_extra_config=None,
+    per_user=False,
+    writable=None,
+    allow_subdir_creation=True,
+    empty_root=False,
+) -> Tuple[TestConfiguredFileSources, str]:
+    if empty_root:
+        tmp, root = "/", None
+    else:
+        tmp, root = setup_root()
     config_kwd = {}
     if include_allowlist:
         config_kwd["symlink_allowlist"] = [tmp]
     file_sources_config = ConfiguredFileSourcesConfig(**config_kwd)
-    plugin = {
+    plugin: Any = {
```

Comment:
```suggestion
    plugin: Dict[str, Any] = {
```

Reply:
Thanks for the catch. Much nicer. I don't know what's up with the CodeQL failure though. That seems unrelated.

Reply:
Definitely, you can ignore those.